src/utils.ts

```diff
@@ -374,23 +376,21 @@ export type TypedDataToPrimitiveTypes<
         : TypedDataToPrimitiveTypes<
             Exclude<TTypedData, K>,
             TAbiParameterKind,
-            TKeyReferences & { [_ in K2['type']]: true }
+            TKeyReferences & { [_ in K2['type'] | K]: true }
           >[K2['type']]
       : // 3. Check if type is array of structs
       K2['type'] extends `${infer TType extends keyof TTypedData &
           string}[${infer Tail}]`
       ? AbiParameterToPrimitiveType<
-          Merge<
```

Comment:
`Merge` overkill. Can just add `name` right on.

Reply:
My guess is that you wanted to preserve if there was no `name` property on the parameter :sweat_smile: But since `name` is a required prop in the typed parameter this should be safe to do.

Reply:
My thinking too!